src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -104,22 +102,41 @@ private bool TryRewriteTagHelper(Block tagBlock, RewritingContext context)
 
             var descriptors = Enumerable.Empty<TagHelperDescriptor>();
 
-            if (IsPotentialTagHelper(tagName, tagBlock))
-            {
-                descriptors = _provider.GetTagHelpers(tagName);
-            }
-
-            // If there aren't any TagHelperDescriptors registered then we aren't a TagHelper
-            if (!descriptors.Any())
+            if (!IsPotentialTagHelper(tagName, tagBlock))
             {
                 return false;
             }
 
+            var currentTagBlockTracker = _tagStack.Count > 0 ? _tagStack.Peek() : null;
+            var tagNameScope = currentTagBlockTracker?.Builder.TagName ?? string.Empty;
+
             if (!IsEndTag(tagBlock))
             {
-                // We're in a begin tag helper block
+                // We're in a begin tag block
+
```

Comment:
actually, this comment and similar ones below correctly identify what gets us to that line of code / the block started on the line before.  the blank links however make it harder to read or associate with what follows.  so remove the distracting blank lines.

if you have additional comments specific to the immediately-following lines of code, start with something like "first, ..." to make the more narrow comment a bit clearer.